Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -42,200 +34,212 @@
-                                } else if (error != null) {
-                                    onError(error);
-                                }
-                            }
-                        });
+                public void onResult(@Nullable List<PaymentMethodNonce> vaultedPaymentMethods, @Nullable Exception error) {
+                    if (vaultedPaymentMethods != null) {
+                        dropInViewModel.setVaultedPaymentMethods(vaultedPaymentMethods);
+                    } else if (error != null) {
+                        onError(error);
                     }
                 }
-            }, getResources().getInteger(android.R.integer.config_shortAnimTime));
+            });
         }
     }
 
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        outState.putBoolean(EXTRA_SHEET_SLIDE_UP_PERFORMED, mSheetSlideUpPerformed);
-        outState.putString(EXTRA_DEVICE_DATA, mDeviceData);
+    private void showSelectPaymentMethodFragment() {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        Fragment fragment = fragmentManager.findFragmentByTag("SELECT_PAYMENT_METHOD");
+        if (fragment == null) {
+            Bundle args = new Bundle();
+            args.putParcelable("EXTRA_DROP_IN_REQUEST", mDropInRequest);
+
+            fragmentManager
+                    .beginTransaction()
+                    .setReorderingAllowed(true)
+                    .add(R.id.fragment_container_view, SelectPaymentMethodFragment.class, args, "SELECT_PAYMENT_METHOD")
+                    .commit();
+        }
+    }
+
+    public void onError(final Exception error) {
+        if (error instanceof AuthenticationException || error instanceof AuthorizationException ||
+                error instanceof UpgradeRequiredException) {
+            getDropInClient().sendAnalyticsEvent("sdk.exit.developer-error");
+        } else if (error instanceof ConfigurationException) {
+            getDropInClient().sendAnalyticsEvent("sdk.exit.configuration-exception");
+        } else if (error instanceof ServerException || error instanceof UnexpectedException) {
+            getDropInClient().sendAnalyticsEvent("sdk.exit.server-error");
+        } else if (error instanceof ServiceUnavailableException) {
+            getDropInClient().sendAnalyticsEvent("sdk.exit.server-unavailable");
+        } else {
+            getDropInClient().sendAnalyticsEvent("sdk.exit.sdk-error");
+        }
+
+        finish(error);
+    }
+
+    private void finishWithDropInResult(DropInResult dropInResult) {
+        getDropInClient().sendAnalyticsEvent("sdk.exit.success");
+        DropInResult.setLastUsedPaymentMethodType(DropInActivity.this, dropInResult.getPaymentMethodNonce());
+        finish(dropInResult.getPaymentMethodNonce(), dropInResult.getDeviceData());
+    }
+
+    private void showPayPal() {
```

Comment:
Nitpick / take it or leave it comment - we could rename this to be more clear that this represents "kicking off / launching the PayPal flow". This and the below method names made me thing of just "showing the PayPal payment selection option", when they actually kick off the flow.

Reply:
Def! Went with `start<FEATURE>Flow` naming scheme.